server.go

```diff
@@ -560,18 +560,24 @@ func (srv *Server) serveDNS(m []byte, w *response) {
 	req := new(Msg)
 	req.setHdr(dh)
 
-	switch srv.MsgAcceptFunc(dh) {
+	switch action := srv.MsgAcceptFunc(dh); action {
 	case MsgAccept:
 		if req.unpack(dh, m, off) == nil {
 			break
 		}
 
 		fallthrough
-	case MsgReject:
+	case MsgReject, MsgRejectNotImplemented:
+		opcode := req.Opcode
 		req.SetRcodeFormatError(req)
+		req.Zero = false
+		if action == MsgRejectNotImplemented {
```

Comment:
Can we just have 2 cases one for MsgReject and One for MsgRejectNotImplemented? Doing an extra if inside the case looks a bit weird (@tmthrgd or did you want it like this?)

Reply:
@miekg We need the if-statement unfortunately because I wrote this switch to rely on `fallthrough`s.